### PR TITLE
Add support for signing tar gzip files

### PR DESCRIPTION
**Pull request by tmat**
This is necessary for our VS Code NPM packages.

On .NET Core we can simply use Tar APIs.
Unfortunately, the APIs are not available on .NET Framework and we still need SignTool to run on .NET Framework (forced by VSSDK msbuild tasks that still do not work on .NET Core). To work around that the change adds a new tool package: Microsoft.Dotnet.Tar that contains a simple .NET Core CLI tool that packs/unpacks tar.gz  files. This tool is then invoked from the .NET Framework msbuild task.

Fixes https://github.com/dotnet/arcade/issues/2915

TODO in future: 
- [ ] Add .tar.gz extension  -- `Path.GetExtension("file.tar.gz")` returns just `.gz` so need to find all places that use the API when comparing `FileExtensionSignInfo` items.
- [ ] To limit risk the repo needs to add to their `eng\Signing.props` file:
```xml
<ItemGroup>
  <ItemsToSign Include="$(ArtifactsPackagesDir)**\*.tgz" />
</ItemGroup>
```

Once we get Roslyn and other repos working we can include these files by default in Arcade.

### CODE REVIEW

**Comment by markwilkie** [comment]
Thanks @tmat!

@chcosta - any comments on this?

cc/ @mmitche

**Comment by chcosta** [comment]
I'll take a look today, there's a lot to look at...

File: Arcade.sln [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ﻿
 Microsoft Visual Studio Solution File, Format Version 12.00
-# Visual Studio Version 16
-VisualStudioVersion = 16.0.29019.234
+# Visual Studio Version 17
+VisualStudioVersion = 17.8.34022.349
```
Comment (chcosta) [inline Arcade.sln:4]: I'm not a huge fan of the requirement on a Preview version of VS, but I don't have enough recent experience to know how impactful / problematic that will be to anyone.

**Review by chcosta — DISMISSED** [review]

File: Arcade.sln (hunk 1 above)
Comment (tmat) [reply Arcade.sln:4]: VS updates this automatically :(

**Comment by mmitche** [comment]
@tmat Microbuild now supports Core, so we may be able to eventually get rid of the Framework targeted signtool.

**Review by chcosta — APPROVED** [review]

**Pull request merged**